File: apps/cargo-tracker/internal/shared/domain/clock.go
package domain

import "time"

// Clock は現在時刻を返す横断ポート。time.Now への直接依存を排し、テストの決定性を確保する。
type Clock interface {
	Now() time.Time
}

// SystemClock は本番用の Clock 実装（time.Now を返す）。
type SystemClock struct{}

// Now は現在時刻を返す。
func (SystemClock) Now() time.Time { return time.Now() }

// FixedClock はテスト用の固定時刻 Clock。
type FixedClock struct {
	Fixed time.Time
}

// Now は固定時刻を返す。
func (c FixedClock) Now() time.Time { return c.Fixed }

File: docs/article/source/typescript/apps/cargo-tracker/src/shared/infrastructure/clock/clock.ts
/**
 * 現在時刻プロバイダ（IT7 1.4）。
 * 未来日ガードの「現在時刻」を DI で差し替え可能にし、Date.now() 直呼びによるテスト不安定を避ける。
 */
export type Clock = () => Date;

/** Clock の DI トークン */
export const CLOCK = Symbol('CLOCK');

/** 既定の Clock（実時刻） */
export const systemClock: Clock = () => new Date();
